src/Config.php

```diff
@@ -2,11 +2,12 @@
 
 namespace Tnapf\Config;
 
+use Tnapf\Config\ConfigProvider\ConfigProvider;
 use Tnapf\Config\Exceptions\InvalidConfigException;
 
 class Config
 {
-    public function __construct(private readonly string $directory)
+    public function __construct(private readonly ConfigProvider $configProvider)
```

Comment:
If you want to use `readonly` modifiers, bump the `php` in `composer.json` to `>=8.1`